nntrainer/layers/layer_node.cpp

```diff
@@ -130,20 +130,46 @@ class SharedFrom : public Name {
  */
 LayerNode::~LayerNode() = default;
 
+/**
+ * @brief get the compute engine property from property string vector
+ *  : default is CPU
+ * @return LayerComputeEngine Enum : CPU, GPU, QNN
+ *
+ */
+ml::train::LayerComputeEngine
+getComputeEngine(const std::vector<std::string> &props) {
+  for (auto &prop : props) {
+    std::string key, value;
+    int status = nntrainer::getKeyValue(prop, key, value);
+    if (nntrainer::istrequal(key, "engine")) {
+      if (nntrainer::istrequal(value, "qnn")) {
+        return ml::train::LayerComputeEngine::QNN;
+      } else if (nntrainer::istrequal(value, "gpu")) {
+        return ml::train::LayerComputeEngine::GPU;
+      }
+    }
+  }
+
+  return ml::train::LayerComputeEngine::CPU;
```

Comment:
If this function is called frequently and may affect the latency, you may use:
```
if (key[0] == 'q')
  return  QNN;
if (key[0] == 'g')
  return GPU;
return CPU;
```

If you want a generalized code (wtih a bit of inefficiency), use EnumStr
```
for (i = 0; i < arraysize(EnumStr); i++) {
  if (istrequal(key, EnumStr[i])
    return EnumList[i];
}
```